src/main/java/org/jenkinsci/plugins/ewm/facets/WorkspaceBrowserFacet.java

```diff
@@ -0,0 +1,54 @@
+package org.jenkinsci.plugins.ewm.facets;
+
+import hudson.model.Fingerprint;
+import jenkins.model.FingerprintFacet;
+import org.jenkinsci.plugins.ewm.model.ExternalWorkspace;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+/**
+ * TODO JAVADOC
+ *
+ * @author Alexandru Somai
+ */
+public class WorkspaceBrowserFacet extends FingerprintFacet {
+
+    public WorkspaceBrowserFacet(@Nonnull Fingerprint fingerprint, long timestamp) {
+        super(fingerprint, timestamp);
+    }
+
+    /**
+     * TODO JAVADOC
+     *
+     * @return
+     */
+    @Restricted(NoExternalUse.class)
+    @SuppressWarnings("unused")
+    public String getWorkspaceLink() {
+        ExternalWorkspace workspace = getWorkspace();
+        if (workspace == null) {
+            return "#";
+        }
+
+        Fingerprint.BuildPtr original = getFingerprint().getOriginal();
+        if (original == null) {
+            return "#";
+        }
+
+        return original.getJob().getAbsoluteUrl() + original.getRun().getSearchUrl() +
```

Comment:
If original run gets deleted, you will have no access to this workspace anymore.
Maybe it makes sense to make the browser independent from the original run. E.g. you can just access workspaces by a link like JENKINS_URL/exws/browse/WORKSPACE_ID